internal/rbd/nodeserver.go

```diff
@@ -224,12 +261,31 @@ func populateRbdVol(
 			return nil, status.Errorf(codes.Internal, "unsupported krbd Feature")
 		}
 		// fallback to rbd-nbd,
-		// ignore the mapOptions and unmapOptions as they are meant for krbd use.
 		rv.Mounter = rbdNbdMounter
 	} else {
 		rv.Mounter = req.GetVolumeContext()["mounter"]
-		rv.MapOptions = req.GetVolumeContext()["mapOptions"]
-		rv.UnmapOptions = req.GetVolumeContext()["unmapOptions"]
+	}
+
+	var (
+		krbdMapOptions   string
+		nbdMapOptions    string
+		krbdUnmapOptions string
+		nbdUnmapOptions  string
+	)
+	krbdMapOptions, nbdMapOptions, err = getMapOptions(req.GetVolumeContext()["mapOptions"])
+	if err != nil {
+		return nil, err
+	}
+	krbdUnmapOptions, nbdUnmapOptions, err = getMapOptions(req.GetVolumeContext()["unmapOptions"])
+	if err != nil {
+		return nil, err
+	}
+	if rv.Mounter == rbdDefaultMounter {
+		rv.MapOptions = krbdMapOptions
+		rv.UnmapOptions = krbdUnmapOptions
+	} else if rv.Mounter == rbdNbdMounter {
+		rv.MapOptions = nbdMapOptions
+		rv.UnmapOptions = nbdUnmapOptions
```

Comment:
move this to a helper function to set map and unmap options?

Reply:
Ok!